### DISCUSSION TITLE
Lifecycle hooks - update container conditionally

### DISCUSSION THREAD

**Discussion by kejdajar** [containrrr/watchtower#847] [Q&A]
Hello, 

is it possible to execute pre/post-check or pre/post-update commands for every container updated by watchtower and also obtain ID or name of the container being updated? 

Secondly, can we use lifecycle hooks to configure whether the specific container should be updated? 

We want to have a custom .sh script that would be responsible for deciding whether the container should be updated or not. We can set up multiple watchtower containers using scopes, but we are looking for other ways how to conditionally update just some of our containers.

**Comment by piksel** [comment]
It should be possible to skip, but doesn't due to #649

**Reply by yrien30** [reply]
You can optain container ID for the container being updated from the shell script using $HOSTNAME I think. Hopefully this pull request gets merged soon. https://github.com/containrrr/watchtower/pull/793 So its possible to skip the containers using preupdate script
